### ISSUE TITLE
Vue 3 support

### ISSUE THREAD

**Issue by andonov85** [FGRibreau/ui-predicate#152]
Hello, **François-Guillaume Ribreau** and thank you for the great library!

We use the library in a project that we plan to migrate to Vue 3 and the question is whether support for Vue 3 is planned in the near future?

Thanks!

**Comment by marclaporte** [comment]
@FGRibreau What is complexity / level of work required here? Is this something a junior developer (in my team) could help with?

Thanks!

**Comment by FGRibreau** [comment]
Hello @marclaporte,

Since there is already an exemple facade for vue2, I think creating a facade for vue3 should not be a big deal yes :)

**Comment by marclaporte** [comment]
@johnsantosDev Show us what you got :-)

**Comment by johnsantosDev** [comment]
Thanks @marclaporte, on it!

**Comment by marclaporte** [comment]
Change of plan (sorry)

@alvinBM is now in charge.

**Comment by marclaporte** [comment]
Closed as done! https://github.com/FGRibreau/ui-predicate/pull/192

Congrats @alvinBM and thank you to @FGRibreau

**Comment by FGRibreau** [comment]
@alvinBM @marclaporte @johnsantosDev Hello! Just a little update, I've indeed merged the PR but lots of things are not working (storybooks, tests, docs) on the vuejs3 package, I've started a refactor to make these parts rights and then I will create a new release.

Thanks a lot for your work!

**Comment by marclaporte** [comment]
@FGRibreau I am sending you positive thoughts on this topic :-)

**Comment by FGRibreau** [comment]
@marclaporte I'm sorry, I don't have the time right now to work on fixing all the missing things in this PR to make it mergeable in ui-predicate, maybe you have available developer bandwidth to do it (fix storybooks, tests & docs generation)?

**Comment by alvinBM** [comment]
> @marclaporte I'm deeply sorry Marc, I don't have the time right now to work on fixing all the missing things in this PR to make it mergeable in ui-predicate, maybe you have available developer bandwidth to do it (fix storybooks, tests & docs generation)?
> 
> (I can guide one of your dev asynchronously though!)

@FGRibreau , I am available to fix the various blocking points, so I'll always come back to you if necessary to move forward more quickly. THANKS

**Comment by marclaporte** [comment]
@FGRibreau Please re-open this task and assign to @alvinBM

**Comment by FGRibreau** [comment]
done @marclaporte !

**Comment by alvinBM** [comment]
Hello @FGRibreau,

I am in the process of fixing the problems related to the doc and the storybooks, but I am currently facing a lot of bugs and perhaps incompatibility especially with the storybooks.

I built the ui-predicate-vue3 package with vite 4.3.2 and vue 3.2.13, so I got a lot of errors related to **parcelRequire** used in `ui-predicate-core`.

I wonder if the current **`ui-predicate-core`** package is compatible with vite? Maybe you can have additional information that can help me move forward faster.

NB: For storybooks I use version 8.0.0

**Comment by Yannick243** [comment]
Hello @FGRibreau,

Here is the PR that fixes the points mentioned below (fixes some bugs, updates Vue3 docs, and adds related Storybooks) [https://github.com/FGRibreau/ui-predicate/pull/204](https://github.com/FGRibreau/ui-predicate/pull/204)

**Comment by marclaporte** [comment]
We can close via https://github.com/FGRibreau/ui-predicate/pull/204 and wait for a new version! 2.0.0?

**Comment by marclaporte** [comment]
@FGRibreau How can we help for the next step? Thanks!

**Comment by FGRibreau** [comment]
@marclaporte sorry for the delay, I'm under heavy workload currently, the 2.0 release is on my todo-list be sure I will complete it in the next weeks :)

**Comment by marclaporte** [comment]
@FGRibreau Is there anything we can do to help?

**Comment by FGRibreau** [comment]
@marclaporte I'm really sorry, I've cofounded [multiple](https://www.getnatalia.com/) [new](https://france-nuage.fr/) [product](https://www.netir.fr/) in the past months and can't find time to go back to ui-predicate.
Would you like to be a co-maintainer ?

**Comment by marclaporte** [comment]
Yes, with my colleagues @alvinBM @Yannick243 @jonnybradley and @kroky

**Comment by FGRibreau** [comment]
Ok, I invited you for now, let's see how it goes first :)

**Comment by marclaporte** [comment]
hehe, OK but I don't code anymore. I focus on my role as a CEO.

**Comment by FGRibreau** [comment]
Duly noted, invited your team then :)

**Comment by jonnybradley** [comment]
Thanks, accepted - just wish i had time to get back into Vue stuff! Soon soon... 🤞

**Comment by Yannick243** [comment]
new PR with enhancements [https://github.com/FGRibreau/ui-predicate/pull/209](https://github.com/FGRibreau/ui-predicate/pull/209)

**Comment by marclaporte** [comment]
@Yannick243 @FGRibreau Anything left here or we can close this?

Thanks!
